### PR TITLE
Deduplicate errors emitted in the eval test stage

### PR DESCRIPTION
**Pull request by saecki**
Fixes the issue described in https://github.com/typst/typst/pull/7738#issuecomment-3801883883

Due to the changes in how diagnostics are printed after https://github.com/typst/typst/pull/7820/commits/e1475583f622f10e065854bf5d1eab1d4c6b7c71 this isn't really noticeable, but is conceptually the consisten thing to do and might still improve the printed message in some edge cases.

### CODE REVIEW

File: tests/src/run.rs [hunk 1]
```diff
@@ -360,35 +361,55 @@ impl<'a> Runner<'a> {
...
     /// Compile a document with the specified target.
-    fn compile<D: TestDocument>(&mut self) -> Option<D> {
+    fn compile<D: TestDocument>(
+        &mut self,
+        evaluated: Warned<SourceResult<Content>>,
+    ) -> Option<D> {
         let Warned { output, warnings } = typst::compile::<D>(&self.world);
-        for warning in &warnings {
+
+        // Conecptually this functions takes the evaluated content as input and
```
Comment (laurmaedje) [inline tests/src/run.rs]:
```suggestion
        // Conceptually this functions takes the evaluated content as input and
```

File: tests/src/run.rs [hunk 2]
```diff
@@ -360,35 +361,55 @@ impl<'a> Runner<'a> {
...
         }
 
-        output.ok()
+        evaluated
     }
 
     /// Compile a document with the specified target.
-    fn compile<D: TestDocument>(&mut self) -> Option<D> {
+    ///
+    /// Conceptually, this functions takes the evaluated content as input and
```
Comment (isuffix) [inline tests/src/run.rs:383]:
```suggestion
    /// Conceptually, this function takes the evaluated content as input and
```

Comment (saecki) [reply tests/src/run.rs:383]: damnit 😄
